src/promptflow/promptflow/executor/_input_assignment_parser.py

```diff
@@ -19,18 +20,44 @@ def parse_value(i: InputAssignment, nodes_outputs: dict, flow_inputs: dict):
     if i.value_type == InputValueType.FLOW_INPUT:
         if i.value not in flow_inputs:
             flow_input_keys = ", ".join(flow_inputs.keys()) if flow_inputs is not None else None
-            raise InputNotFound(message=f"{i.value} is not found from flow inputs '{flow_input_keys}'")
+            raise InputNotFound(
+                message_format=(
+                    "The input '{input_name}' is not found from flow inputs '{flow_input_keys}'. "
+                    "Please check the input name and try again."
+                ),
+                input_name=i.value,
+                flow_input_keys=flow_input_keys,
+            )
         return flow_inputs[i.value]
     if i.value_type == InputValueType.NODE_REFERENCE:
         if i.section != "output":
-            raise UnsupportedReference(f"Unsupported reference {i.serialize()}")
+            raise UnsupportedReference(
+                message_format=(
+                    "The section '{reference_section}' of reference is currently unsupported. "
+                    "Please specify the output part of the node '{reference_node_name}'."
+                ),
+                reference_section=i.section,
+                reference_node_name=i.value,
+            )
         if i.value not in nodes_outputs:
             node_output_keys = ", ".join(nodes_outputs.keys()) if nodes_outputs is not None else None
             raise InputNotFoundFromAncestorNodeOutput(
-                message=f"{i.value} is not found from ancestor node output '{node_output_keys}'"
+                message_format=(
+                    "The input '{input_name}' is not found from ancestor node outputs '{node_output_keys}'. "
```

Comment:
1. node_output_keys is list , we do not need single quote it.
2. any template

Reply:
1. it is string.
2. updated.

Reply:
1. change it to list